projects/js-packages/charts/src/components/pie-semi-circle-chart/pie-semi-circle-chart.tsx

```diff
@@ -302,11 +303,23 @@ const PieSemiCircleChartInternal: FC< PieSemiCircleChartProps > = ( {
 	);
 };
 
-const PieSemiCircleChart: FC< PieSemiCircleChartProps > = props => (
+const PieSemiCircleChartBase: FC< PieSemiCircleChartProps > = props => (
 	<ChartProvider>
 		<PieSemiCircleChartInternal { ...props } />
 	</ChartProvider>
 );
 
-PieSemiCircleChart.displayName = 'PieSemiCircleChart';
+PieSemiCircleChartBase.displayName = 'PieSemiCircleChart';
+
+// Create composition type
+type PieSemiCircleChartComponent = FC< PieSemiCircleChartProps > & {
```

Comment:
I'm thinking if we type `attachSubComponents` properly then we probably could save a couple of line not to define the composition types in all the chart types?